File: demo/insecure_demo.java
import java.io.FileInputStream;
import java.io.ObjectInputStream;
import java.security.MessageDigest;

public class InsecureDemo {
    private static final String PASSWORD = "HardcodedPassword123";

    public void loadUser(Database db, String userInput) {
        String query = "SELECT * FROM users WHERE id=" + userInput;
        db.query(query);
    }

    public void readPayload(String path) throws Exception {
        ObjectInputStream input = new ObjectInputStream(new FileInputStream(path));
        input.readObject();
    }

    public byte[] weakHash(byte[] data) throws Exception {
        MessageDigest digest = MessageDigest.getInstance("SHA1");
        return digest.digest(data);
    }
}

File: demo/insecure_demo.js
const crypto = require("crypto");
const fs = require("fs");
const { exec } = require("child_process");

const apiKey = "demo-client-secret-12345";

function loadUser(db, userInput) {
  const query = "SELECT * FROM users WHERE id=" + userInput;
  return db.query(query);
}

function renderProfile(target, userContent) {
  target.innerHTML = userContent;
}

function runTask(userInput) {
  exec("ls " + userInput);
}

function weakDigest(value) {
  return crypto.createHash("sha1").update(value).digest("hex");
}

function downloadAvatar() {
  return fetch("http://api.example.com/users");
}

function readUserFile(req) {
  return fs.readFileSync(req.query.path, "utf8");
}
